hikari/polls.py

```diff
@@ -0,0 +1,318 @@
+
+__all__: typing.Sequence[str] = (
+    "PollMedia",
+    "PollAnswer",
+    "PollResult",
+    "PollAnswerCount",
+    "PollLayoutType",
+    "PartialPoll",
+    "PollCreate",
+    "PollObject",
+)
+
+import typing
+
+import attrs
+
+from hikari.emojis import Emoji
+from hikari.internal import attrs_extensions
+from hikari.internal import enums
+
+if typing.TYPE_CHECKING:
+    import datetime
+
+
+def _ensure_optional_emoji(emoji: typing.Optional[typing.Union[str, Emoji]]) -> Emoji | None:
+    """Ensure the object is a [hikari.emojis.Emoji][]."""
+    if emoji is not None:
+        return Emoji.parse(emoji) if isinstance(emoji, str) else emoji
+    return None
+
+
+@attrs_extensions.with_copy
+@attrs.define(hash=False, kw_only=True, weakref_slot=False)
+class PollMedia:
+    """Common object backing a poll's questions and answers."""
+
+    text: typing.Optional[str] = attrs.field(default=None, repr=True)
+    """The text of the element, or [`None`][] if not present."""
+
+    emoji: typing.Optional[Emoji] = attrs.field(default=None, repr=True)
+    """The emoji of the element, or [`None`][] if not present."""
+
+
+@attrs_extensions.with_copy
+@attrs.define(hash=False, kw_only=True, weakref_slot=False)
+class PollAnswer:
+    """Represents an answer to a poll."""
+
+    answer_id: int = attrs.field(repr=True)
+    """The ID that labels this answer."""  # Is this user-settable?
+
+    poll_media: PollMedia = attrs.field(repr=True)
+    """The [media][hikari.polls.PollMedia] associated with this answer."""
+
+
+@attrs_extensions.with_copy
+@attrs.define(hash=False, kw_only=True, weakref_slot=False)
+class PollResult:
+    """Represents a poll result."""
+
+    is_finalized: bool = attrs.field(repr=True)
+    """Whether the poll is finalized and the votes are precisely counted."""
+
+    answer_counts: typing.Sequence[PollAnswerCount] = attrs.field(repr=True)
+    """The counts for each answer."""
+
+
+@attrs_extensions.with_copy
+@attrs.define(hash=False, kw_only=True, weakref_slot=False)
+class PollAnswerCount:
+    """Represents the count of a poll answer."""
+
+    answer_id: int = attrs.field(repr=True)
+    """The ID of the answer."""
+
+    count: int = attrs.field(repr=True)
+    """The number of votes for this answer."""
+
+    me_voted: bool = attrs.field(repr=True)
+    """Whether the current user voted for this answer."""
+
+
+class PollLayoutType(int, enums.Enum):
+    """Layout of a poll."""
+
+    DEFAULT = 1
+    """The default layout of a poll."""
+
+
+class PartialPoll:
+    """Base class for all poll objects."""
+
+    __slots__: typing.Sequence[str] = ("_question", "_answers", "_allow_multiselect", "_layout_type", "_counter")
+
+    def __init__(self, question: str, allow_multiselect: bool, layout_type: typing.Union[int, PollLayoutType]):
+        self._question = PollMedia(text=question)  # Only text is supported for question
+        self._allow_multiselect = allow_multiselect
+        self._layout_type = layout_type
```

Comment:
This object is a bit too confusing. It should be more like embed (or it should be moved as a special endpoint class)

Reply:
Considering it's a certified Discord moment™️ by having a separate poll object and poll creation object, I have absolutely no idea how to implement the two classes elegantly without copy-pasting code. Hence the hacky `PartialPoll`.

Is it against stuffs like `PartialMessage`, `PartialGuild`, etc.? Yes. Do I have any other ideas? No, sadly.